backend/de.metas.contracts/src/main/java/de/metas/contracts/commission/commissioninstance/services/hierarchy/CommissionHierarchyFactory.java

```diff
@@ -36,20 +36,36 @@
  * #L%
  */
 
-/** Crates a hierarchy of BPartners by recursively following {@code C_BPartner.C_BPartner_SalesRep_ID} references. */
+/**
+ * Crates a hierarchy of BPartners by recursively following {@code C_BPartner.C_BPartner_SalesRep_ID} references.
+ */
 @Service
 public class CommissionHierarchyFactory
 {
-	// very crude but simple implementation; expand and make more efficient as needed
-	public Hierarchy createFor(@NonNull final BPartnerId bPartnerId)
+	@NonNull
```

Comment:
```suggestion
	// very crude but simple implementation; expand and make more efficient as needed
	@NonNull
```

sorry to say, but imho it's still crude :-) as we load one bpartner after the other, instead of finding ways to load them all at once